stdlib/universe/mode.go

```diff
@@ -0,0 +1,668 @@
+
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+	case flux.TFloat:
+		floatMode := make(map[float64]int64)
+		return tbl.Do(func(cr flux.ColReader) error {
+			l := cr.Len()
+			err := t.doFloat(cr, tbl, floatMode, l, numNil, builder, colIdx)
+
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+	case flux.TString:
+		stringMode := make(map[string]int64)
+		return tbl.Do(func(cr flux.ColReader) error {
+			l := cr.Len()
+			err := t.doString(cr, tbl, stringMode, l, numNil, builder, colIdx)
+
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+	case flux.TTime:
+		timeMode := make(map[execute.Time]int64)
+		return tbl.Do(func(cr flux.ColReader) error {
+			l := cr.Len()
+			err := t.doTime(cr, tbl, timeMode, l, numNil, builder, colIdx)
+
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+	}
+
+	if err := execute.AppendKeyValues(tbl.Key(), builder); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *modeTransformation) doString(cr flux.ColReader, tbl flux.Table, stringMode map[string]int64, l int, numNil int64, builder execute.TableBuilder, colIdx int) error {
+	j := execute.ColIdx(t.column, tbl.Cols())
+	for i := 0; i < l; i++ {
+		if cr.Strings(j).IsNull(i) {
+			numNil++
+			continue
+		}
+		v := cr.Strings(j).ValueString(i)
+		stringMode[v]++
+	}
+
+	max, total := int64(0), int64(0)
+	for val := range stringMode {
+		if stringMode[val] > max {
+			max, total = stringMode[val], 1
+		} else if stringMode[val] == max {
+			total++
+		}
+	}
+
+	storedVals := make([]string, 0, total)
+	for val := range stringMode {
+		if stringMode[val] == max {
+			storedVals = append(storedVals, val)
+		}
+	}
+
+	/*
+		// byte-slice method:
+		for k := range stringMode {
+			if stringMode[k] > max {
+				storedVals = storedVals[:0]
+				storedVals = append(storedVals, k)
+				max = stringMode[k]
+			} else if stringMode[k] == max {
+				storedVals = append(storedVals, k)
+			}
+		}
+	*/
+
+	if numNil > max {
+		if err := builder.AppendNil(colIdx); err != nil {
+			return err
+		}
+	} else if len(storedVals) == len(stringMode) && (numNil == 0 || numNil == max) {
+		if err := builder.AppendNil(colIdx); err != nil {
+			return err
+		}
+	} else if numNil == max {
+		sort.Strings(storedVals)
+		for j := range storedVals {
+			if err := builder.AppendString(colIdx, storedVals[j]); err != nil {
+				return err
+			}
+		}
+		if err := builder.AppendNil(colIdx); err != nil {
+			return err
+		}
+	} else {
+		sort.Strings(storedVals)
+		for j := range storedVals {
+			if err := builder.AppendString(colIdx, storedVals[j]); err != nil {
+				return err
+			}
+		}
+	}
+
+	if err := execute.AppendKeyValues(tbl.Key(), builder); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *modeTransformation) doBool(cr flux.ColReader, tbl flux.Table, boolMode map[bool]int64, l int, numNil int64, builder execute.TableBuilder, colIdx int) error {
+	j := execute.ColIdx(t.column, tbl.Cols())
```

Comment:
I think you can just use `colIdx` because the column has already been determined.

Reply:
I get a type conversion panic if I set j to be colIdx.

Reply:
It looks like the reason is because you use `colIdx` twice above so you lose the initial assignment. I think you should resolve that so that you can pass both the source column and the output column rather than using `ColIdx` to determine the column again here.